lib/rspec/core/formatters/exception_presenter.rb

```diff
@@ -31,8 +31,37 @@ def colorized_message_lines(colorizer=::RSpec::Core::Formatters::ConsoleCodes)
           end
         end
 
-        def formatted_backtrace
-          backtrace_formatter.format_backtrace(exception_backtrace, example.metadata)
+        def formatted_backtrace(exception=@exception)
+          lines = backtrace_formatter.format_backtrace((exception.backtrace || []), example.metadata)
+
+          lines + formatted_cause(exception)
+        end
+
+        if RSpec::Support::RubyFeatures.supports_exception_cause?
+          def formatted_cause(exception)
+            last_cause = exception_cause(exception)
+            cause = []
+
+            if exception.cause
+              cause << '------------------'
+              cause << '--- Caused by: ---'
+              cause << "#{exception_class_name(last_cause)}:" unless exception_class_name(last_cause) =~ /RSpec/
+
+              encoded_string(last_cause.message.to_s).split("\n").each do |line|
+                cause << "  #{line}"
+              end
+
+              cause << ("  " + backtrace_formatter.format_backtrace(last_cause.backtrace, example.metadata).first)
```

Comment:
Performance wise it'd be better to interpolate the `" "` and the backtrace, rather than + simple because it avoids an extra instantiation